### ISSUE TITLE
Duplicates in AvailableNetworkList

### ISSUE THREAD

**Issue by DircoDT** [jorgebv/windows-wifi-api#7]
The native API often returns duplicate networks for some reason - identical except for the profileName (which is blank for one and equal to the SSID for the other.)

Below is a short way to remove the blank-profileName-duplicate from the returned listing:

```
private Wlan.WlanAvailableNetwork[] GetAvailableNetworkList(Wlan.WlanGetAvailableNetworkFlags flags)
{
    IntPtr availNetListPtr;
    Wlan.ThrowIfError(
        Wlan.WlanGetAvailableNetworkList(client.clientHandle, info.interfaceGuid, flags, IntPtr.Zero, out availNetListPtr));
    try
    {
        return ConvertAvailableNetworkListPtr(availNetListPtr)
                    .GroupBy(w => Encoding.ASCII.GetString(w.dot11Ssid.SSID, 0, (int) w.dot11Ssid.SSIDLength))
                    .Select(g => g.OrderByDescending(i => i.profileName).First())
                    .ToArray(); // Group by SSID, then sort the groups by profileName (which moves the blank string to the bottom) and keep only the first one.
    }
    finally
    {
        Wlan.WlanFreeMemory(availNetListPtr);
    }
}
```

There are [faster ways](http://stackoverflow.com/questions/1300088/distinct-with-lambda), but they require more code (not in-line like the above.)

**Comment by jorgebv** [comment]
Thank you for your detailed issue reports! I've taken a look at them and plan to address the others soon.

Rather than random duplicates, the networks you are seeing twice should be networks in range for which you also have a configured profile. Can you confirm this is what you are seeing? You can check your configured profiles with the cmd command "netsh wlan show profiles"

Although I can't really think of a use case for wanting these duplicates, I want this API to remain a thin wrapper over the native APIs. It's conceivable that someone wants these duplicates and removing them from the results might hamper the usefulness.

**Comment by DircoDT** [comment]
Yes, the duplicates are not random and only occur for the networks that have an existing profile on my computer (access points I've connected to before.)

I can't think of a use case for having duplicates either, but you are right that it's not the wrapper's job to remove them, so you can close this issue.

Maybe just mention the duplicates in a remark above that wrapper function (or in a usage sample somewhere), in case others want to filter it out too.
